src/github.com/getlantern/detour/detect.go

```diff
@@ -0,0 +1,65 @@
+package detour
+
+import (
+	"bytes"
+	"net"
+	"syscall"
+)
+
+type Detector struct {
+	CheckConn    func(net.Conn) bool
+	CheckError   func(error) bool
+	CheckContent func([]byte) bool
+}
+
+var detectors = make(map[string]*Detector)
+
+var iranRedirectAddr = "10.10.34.34:80"
+
+func init() {
+	http403 := []byte("HTTP/1.1 403 Forbidden")
+	iranIFrame := []byte(`<iframe src="http://10.10.34.34`)
+	// see tests and https://github.com/getlantern/lantern/issues/2099#issuecomment-78015418
+	// for the facts behind detection rules for Iran
+	detectors["IR"] = &Detector{
+		CheckConn: func(c net.Conn) bool {
+			if ra := c.RemoteAddr(); ra != nil {
+				return ra.String() == iranRedirectAddr
+			}
+			return false
+		},
+		CheckContent: func(b []byte) bool {
+			return bytes.HasPrefix(b, http403) && bytes.Contains(b, iranIFrame)
```

Comment:
> Iran doesn't hijack content for https connection at all

Cool.  That might be worth a comment here.

> In general, are there any way to hijack content in a https session?

Yes.  Iran in particular is pushing people to use their "National Browser" which will trust stuff that's been MITM'ed by the government - http://www.iranhumanrights.org/2014/11/internet-reportthe-national-browser/